internal/test/captor.go

```diff
@@ -0,0 +1,13 @@
+package test
+
+import "github.com/stretchr/testify/mock"
+
+type Captor struct {
+	Args mock.Arguments
+}
+
+func (c *Captor) Capture() func(mock.Arguments) {
+	return func(args mock.Arguments) {
+		c.Args = args
+	}
+}
```

Comment:
Testify's mocking library allows you to hook into the mock's setup by supplying a func(mock.Arguments) to Mock.Run(). In the docs its used for printing etc. but here we're keeping a copy of the arguments so that we can poke at them later.